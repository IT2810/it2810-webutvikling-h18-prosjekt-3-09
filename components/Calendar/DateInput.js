import React from "react";
import moment from "moment";

import { View } from "react-native";
import { FormLabel, Text } from "react-native-elements";

export default ({ label, onPress, value }) => (
  <View
    style={{
      marginVertical: 8,
      flexDirection: "row",
      alignItems: "baseline",
      justifyContent: "space-between"
    }}
  >
    <FormLabel>{label}</FormLabel>
    <Text
      onPress={onPress}
      style={{
        color: "grey",
        fontSize: 14,
        padding: 8,
        borderWidth: 1,
        borderColor: "black"
      }}
    >
      {value ? moment(value).format("YYYY. MMMM DD") : "not set"}
    </Text>
  </View>
);
