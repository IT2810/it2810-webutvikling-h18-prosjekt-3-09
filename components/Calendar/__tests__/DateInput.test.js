import React from "react";
import renderer from "react-test-renderer";
import DateInput from "../DateInput";
import moment from "moment";
import { FormLabel, Text } from "react-native-elements";

describe("DateInput snapshot", () => {
  let tree;
  const value = new Date();
  const label = "Test";
  beforeAll(() => {
    tree = renderer.create(
      <DateInput onPress={jest.fn()} label={label} value={value} />
    );
  });

  test("renders correctly", () => {
    expect(tree).toMatchSnapshot();
  });

  test("display label", () => {
    expect(
      tree.toTree().rendered.props.children.find(e => e.type === FormLabel)
        .props.children
    ).toBe(label);
  });

  test("display date", () => {
    expect(
      tree.toTree().rendered.props.children.find(e => e.type === Text).props
        .children
    ).toBe(moment(value).format("YYYY. MMMM DD"));
  });
});
