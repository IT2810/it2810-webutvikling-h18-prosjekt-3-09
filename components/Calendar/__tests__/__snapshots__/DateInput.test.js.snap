// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput snapshot renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "baseline",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 8,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {},
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "color": "#86939e",
              "fontSize": 15,
              "fontWeight": "bold",
              "marginBottom": 1,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 15,
            },
            undefined,
            undefined,
          ],
        ]
      }
    >
      Test
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[MockFunction]}
    style={
      Array [
        Object {},
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "borderColor": "black",
          "borderWidth": 1,
          "color": "grey",
          "fontSize": 14,
          "padding": 8,
        },
      ]
    }
  >
    2018. October 15
  </Text>
</View>
`;
